src/Evolution/DiscontinuousGalerkin/Actions/InternalMortarDataImpl.hpp

```diff
@@ -230,53 +229,110 @@ void internal_mortar_data_impl(
            "have been. Direction: "
                << direction);
 
-    // We point the Variables inside the DataVector because the DataVector
-    // will be moved later on so we want it owning its data
-    DataVector packaged_data_buffer{
+    const size_t total_face_size =
         face_mesh.number_of_grid_points() *
-        Variables<mortar_tags_list>::number_of_independent_components};
-    Variables<mortar_tags_list> packaged_data{packaged_data_buffer.data(),
-                                              packaged_data_buffer.size()};
+        Variables<mortar_tags_list>::number_of_independent_components;
+    Variables<mortar_tags_list> packaged_data{};
+
+    // This is the case where we only have one neighbor in this direction, so we
+    // may or may not have to do any projection. If we don't have to do
+    // projection, then we can use the local_mortar_data itself to calculate the
+    // dg_package_data. However, if we need to project, then we hae to use the
+    // mortar_projection_buffer that was passed in.
+    if (neighbors_in_direction.size() == 1) {
+      const auto& neighbor = *neighbors_in_direction.begin();
+      const auto mortar_id = std::pair{direction, neighbor};
+      const auto& mortar_mesh = mortar_meshes.at(mortar_id);
+      const auto& mortar_size = mortar_sizes.at(mortar_id);
 
-    const double max_abs_char_speed_on_face = detail::dg_package_data<System>(
+      // Have to use mortar_projection_buffer
+      if (Spectral::needs_projection(face_mesh, mortar_mesh, mortar_size)) {
+        // The face mesh will be assigned below along with ensuring the size of
+        // the mortar data is correct
+        packaged_data.set_data_ref(mortar_projection_buffer->data(),
+                                   total_face_size);
+      } else {
+        // Can use the local_mortar_data
+        auto& local_mortar_data_opt =
+            mortar_data_ptr->at(mortar_id).local_mortar_data();
+        // If this isn't the first time, set the face mesh
+        if (LIKELY(local_mortar_data_opt.has_value())) {
+          local_mortar_data_opt->first = face_mesh;
+        } else {
+          // Otherwise we need to initialize the pair. If we don't do this, then
+          // the DataVector will be non-owning which we don't want
+          local_mortar_data_opt =
+              std::optional{std::pair{face_mesh, DataVector{}}};
+        }
+
+        // Always set the time step ID
+        mortar_data_ptr->at(mortar_id).time_step_id() = temporal_id;
+
+        auto& local_mortar_data = local_mortar_data_opt->second;
+
+        // Do a destructive resize to account for potential p-refinement
+        local_mortar_data.destructive_resize(total_face_size);
+
+        ASSERT(local_mortar_data.size() == total_face_size,
+               "AhHHhhhhHHHHhHhhhHHh");
```

Comment:
Perhaps be a bit more descriptive here?  We can probably just assume that the `destructive_resize` method works, though.

Reply:
Oops 😅 forgot to take that out...